inc/knowbaseitem_category.class.php

```diff
@@ -0,0 +1,285 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ *  Class KnowbaseItem_Category
+ *
+ *  @author Romain Biot <rbiot@teclib.com>
+ *
+ *  @since 9.5.6
+ */
+class KnowbaseItem_Category extends CommonDBRelation {
+
+   // From CommonDBRelation
+   static public $itemtype_1          = 'KnowbaseItem';
+   static public $items_id_1          = 'knowbaseitems_id';
+   static public $itemtype_2          = 'KnowbaseItemCategory';
+   static public $items_id_2          = 'Knowbaseitemcategories_id';
+   static public $checkItem_2_Rights  = self::HAVE_VIEW_RIGHT_ON_ITEM;
+
+   // From CommonDBTM
+   public $dohistory = true;
+
+   static $rightname = 'knowbase';
+
+   function canPurgeItem() {
+      return Session::haveRight(static::$rightname, UPDATE);
+   }
+
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0) {
+      self::showForItem($item, $withtemplate);
+      return true;
+   }
+
+   static function getItems(CommonDBTM $item, $start = 0, $limit = 0, $used = false) {
+      global $DB;
+
+      $kbi_cat_table = self::getTable();
+
+      $criteria = [
+         'FROM'      => [$kbi_cat_table],
+         'FIELDS'    => [$kbi_cat_table => '*'],
+         'INNER JOIN' => [
+            'glpi_knowbaseitems' => [
+               'ON'  => [
+                  $kbi_cat_table        => 'knowbaseitems_id',
+                  'glpi_knowbaseitems' => 'id'
+               ]
+            ]
+         ],
+         'WHERE'     => [],
+         'GROUPBY'   => [
+            $kbi_cat_table . '.id'
+         ]
+      ];
+      $where = [];
+
+      $items_id  = (int)$item->getField('id');
```

Comment:
1. `getField` method should not be used anymore (or only in a display context).
2. `id` field should already be casted as integer by the mysqli driver.
```suggestion
      $items_id  = $item->fields['id'];
```